### PR TITLE
fix error responses and use json body in posts

### PR DESCRIPTION
**Pull request by kradical**

### CODE REVIEW

File: meteor/views.py [hunk 1]
```diff
@@ -23,74 +28,94 @@ def signin(request):
...
-    new = User(first_name=first, last_name=last, organization=organization, email=email)
-    new.set_password(password)
-    new.save()
-    return HttpResponseRedirect('/login')
+    try:
+        new = User(first_name=first, last_name=last, organization=organization, email=email)
+        new.set_password(password)
+        new.save()
+    except IntegrityError as e:
+        return JsonResponse({'error': 'REGISTRATION_ERROR', 'message': str(e)}, status=400)
```
Comment (kradical) [inline meteor/views.py]: This definitely catches when a user already exists, but I'm not sure what other types of errors can be thrown here. We could do a general except but.. yeah

File: meteor/models.py [hunk 2]
```diff
@@ -1,14 +1,35 @@
...
 
-class Earthquake(models.Model):
+class DisasterMap(models.Model):
+    """
+    A general disaster map created by a user. Each map has many resources
+    """
+    name = models.CharField(max_length=120)
+    created_by = models.ForeignKey('User', on_delete=models.CASCADE)
+    created_time = models.DateTimeField()
+    cluster_activated = models.BooleanField(default=False)
```
Comment (ProgrammingJoe) [inline meteor/models.py]: So an EarthquakeMap would be a sub class of the Disaster Map?

Comment (ProgrammingJoe) [inline meteor/models.py]: Completely missed the DisasterMap parameter. Makes sense.

File: meteor/models.py [hunk 3]
```diff
@@ -36,12 +57,14 @@ class User(AbstractBaseUser, PermissionsMixin):
     APPROVED = 'AP'
     TRUSTED = 'TR'
     ADMIN = 'AD'
+    PRIVILEGES = ((UNAPPROVED, 'Unapproved'), (APPROVED, 'Approved'), (TRUSTED, 'Trusted'), (ADMIN, 'Admin'))
+
+    email = models.EmailField(max_length=120, unique=True)
```
Comment (ProgrammingJoe) [inline meteor/models.py]: Moved to the top because unique=True?

File: meteor/models.py [hunk 4]
```diff
@@ -1,14 +1,35 @@
...
+    """
+    A resource on a map. Each resource belongs to one map.
+    """
+    name = models.CharField(max_length=120)
+    entity_type = models.CharField(max_length=120)
+    
+    quanitity = models.FloatField()
+    unit = models.CharField(max_length=120)
+    
+    location = models.PointField()
```
Comment (ProgrammingJoe) [inline meteor/models.py]: At some point we will likely also want to enter longitude and latitude manually I imagine. Other fields of this model will probably change a bit too as we gather more requirements.

Comment (kradical) [inline meteor/models.py]: a pointfield is a lat and long pair! you can create one like this ```resource.location = GEOSGeometry("POINT(0.0 0.0)")``` There might be other ways to do it but that is the one that I found!

File: meteor/views.py (hunk 1 above)
Comment (ProgrammingJoe) [inline meteor/views.py]: I think this is fine. Being general with the response from user input is good. TIL in SENG 426

File: meteor/views.py [hunk 5]
```diff
@@ -23,74 +28,94 @@ def signin(request):
...
-        'meteor/register.html'
-    )
-
-def disaster(request):
-    return render(
-        request,
-        'meteor/disaster.html',
-        {'create_earthquake_form': CreateEarthquake()}
-    )
+    return HttpResponse("OK")
```
Comment (ProgrammingJoe) [inline meteor/views.py]: OK

File: meteor/models.py (hunk 3 above)
Comment (kradical) [inline meteor/models.py]: yeah, easy to see the pk

**Pull request merged**